bin/spark-class

```diff
@@ -77,6 +77,10 @@ set +o posix
 CMD=()
 DELIM=$'\n'
 CMD_START_FLAG="false"
+
+temp_file=$(mktemp)
```

Comment:
Is it possible to avoid creating a temp file? Maybe wrapping the rest of the file after L84 into another function say run_command, so we have something like `build_command "$@" | run_command` ?